src/util.ts

```diff
@@ -132,6 +152,91 @@ export class Util {
 
     return `https://discordapp.com/api/oauth2/authorize?client_id=${clientId}&scope=bot&permissions=${perms}`;
   }
+
+  public static async GetMxidFromName(intent: Intent, name: string, channelMxids: string[]) {
+    if (name[0] === "@" && name.includes(":")) {
+        return name;
+    }
+    const client = intent.getClient();
+    const matrixUsers = {};
+    let matches = 0;
+    await Promise.all(channelMxids.map((chan) => {
+      // we would use this.bridge.getBot().getJoinedMembers()
+      // but we also want to be able to search through banned members
+      // so we gotta roll our own thing
+      return client._http.authedRequestWithPrefix(
```

Comment:
Eek, this looks expensive :(

Reply:
It doesn't get called frequently, so i thought it wasn't too big a concern? Any suggestion ideas, though? (we did discuss exactly this in the matrix room once o.o)